packages/reporter/src/main.tsx

```diff
@@ -60,21 +40,23 @@ class Reporter extends Component<ReporterProps> {
   }
 
   render () {
-    const { appState } = this.props
-
     return (
       <div className='reporter'>
-        <Header appState={appState} statsStore={this.props.statsStore} />
-        <Runnables
-          appState={appState}
-          error={this.props.error}
-          runnablesStore={this.props.runnablesStore}
-          scroller={this.props.scroller}
-          spec={this.props.spec}
-        />
-        <ForcedGcWarning
-          appState={appState}
-          events={this.props.events}/>
+        {this.props.specs.map((spec) => (
```

Comment:
We probably need a backward compatibility mode, for existing e2e which should not be changed. That's what `runMode` was doing.

Reply:
Actually this *is* backwards compatible - I updated e2e to always work with an array of specs. At this point it will always be an array with 1 spec in it, like `[{ relative: '...', absolute: '...' }]`. For all tests, it will be `[{ relative: '_all' }]` (or whatever it is).

The idea here was to avoid having two modes, and just have one. Then, eventually, e2e can be updated to support multiple specs too (eg, either using a checkbox in the GUI or passing multiple specs to the `--spec` arg).

Does this make sense? We can jump on a call to discuss this more, I went over this design idea w/ @chrisbreiding and @JessicaSachs. See here for a more detailed breakdown: https://cypress-io.atlassian.net/browse/CT-164?focusedCommentId=12692

What do you think?

Reply:
The main thing – that it requires a new build, it has a new prop, method names and a logic inside which can be a pain in terms of releasing binary + test runner. I am not aware of this, but in terms of we are doing a giant PR that has a lot of changes this can be hard to keep track of the changes into e2e, and we can not ship these changes without binary release